solutions/USACO jan19-grass-plantin.mdx

```diff
@@ -0,0 +1,102 @@
+---
+id: usaco-849
+source: USACO Silver 2019 January
+title: Grass Planting
+author: Aadit Ambadkar
+---
+
+## Solution 1 (Smart & Intended Solution)
+
+$\mathcal{O}(N)$ using key insight.
+
+Let ```adj[i]``` represent the number of paths adjacent to node $i$ in the graph. Then, considering that node, and that node alone, you would need exactly ```adj[i]+1``` numbers to fill it. This is because you would need ```adj[i]``` for each of the adjacent nodes, and $1$ for the node itself. So, our answer is the maximum of all the adjacents, plus 1. My Java code is below.
```

Comment:
```suggestion
Let ```adj[i]``` represent the number of paths adjacent to node $i$ in the graph. Then, considering that node, and that node alone, you would need exactly ```adj[i]+1``` numbers to fill it. This is because you would need ```adj[i]``` for each of the adjacent nodes, and $1$ for the node itself. So, our answer is the maximum of all the adjacents, plus $1$. 
```

For inline code you can use single `'s. for example:
```
Let `adj[i]` represent ...
```

Also when writing variables we prefer to stick with latex - in this case you would use "$\texttt{adj}[i]$" over "\`adj[i]\`"

Finally, by "adj" do you mean the "degree" or "deg"?